doc/guides/kconfig/index.rst

```diff
@@ -1,918 +1,15 @@
 .. SPDX-License-Identifier: Apache-2.0
 
-.. _kconfig_tips_and_tricks:
+.. _kconfig:
 
-Kconfig - Tips and Best Practices
-#################################
+Kconfig
+#######
 
-.. contents::
-    :local:
-    :depth: 2
+.. toctree::
```

Comment:
We could use a short introduction to configuration and Kconfig here, before sending folks off into the detailed topics listed.  Something like this perhaps:
```
There are many different configurations possible for the Zephyr
kernel and subsystems to adapt them for specific application and
platform needs. Many configuration choices can be made without
actually changing the source code, by first configuring the kernel
and subsystem options and then building the source with those options.

Zephyr uses the same Kconfig system used by the Linux Kernel.
Kconfig files (organized in a tree structure within the Zephyr code tree)
describe build-time configuration options called *symbols*, how they’re
grouped into menus and sub-menus, and dependencies between
them that determine what configurations are valid.

The following sections go into details about Kconfig use within the
Zephyr project, how to specify configuration options, tips and
notes about Kconfig, and more.
```

Reply:
+1, I think we can even emphasise that this is one of the asset of zephyr to achieve low footprint goal

Reply:
Added a similar intro now, along with some misc. info like the header name. Check that it looks okay.